src/til/ut_til/SomeTests.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include "WexTestClass.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+class SomeTests
+{
+    TEST_CLASS(SomeTests);
+
+    TEST_METHOD(Fill)
+    {
+        til::some<int, 4> s;
+
+        const auto val = 12;
+        s.fill(val);
+
+        VERIFY_ARE_EQUAL(s.max_size(), s.size());
+
+        for (const auto& i : s)
+        {
+            VERIFY_ARE_EQUAL(val, i);
+        }
+    }
+
+    TEST_METHOD(Swap)
+    {
+        til::some<int, 4> a;
+        til::some<int, 4> b;
+
+        const auto aVal = 900;
+        a.fill(900);
+
+        const auto bVal = 45;
+        b.push_back(45);
+
+        const auto aSize = a.size();
+        const auto bSize = b.size();
+
+        a.swap(b);
+
+        VERIFY_ARE_EQUAL(aSize, b.size());
+        VERIFY_ARE_EQUAL(bSize, a.size());
+
+        VERIFY_ARE_EQUAL(bVal, a[0]);
+
+        for (const auto& i : b)
+        {
+            VERIFY_ARE_EQUAL(aVal, i);
+        }
+    }
+
+    TEST_METHOD(Size)
+    {
+        til::some<int, 2> c;

```

Comment:
Build appears unhappy about the signed/unsigned mismatches comparing raw integers against `size`

I get it, but the compiler could have been smarter.

Reply:
got it.